File: apps/ai-service/app/providers/routing.py
from app.models.schemas import Authority, ReportCategory

# Preferred authority type for each report category, best match first.
CATEGORY_TO_AUTHORITY_TYPES: dict[ReportCategory, list[str]] = {
    ReportCategory.flood: ["fire", "municipal", "utilities"],
    ReportCategory.accident: ["police", "health", "fire"],
    ReportCategory.infrastructure: ["municipal", "utilities"],
    ReportCategory.gov_delay: ["municipal"],
    ReportCategory.safety: ["police"],
    ReportCategory.other: ["municipal"],
}


def suggest_authority_id(
    category: ReportCategory, authorities: list[Authority]
) -> str | None:
    """Pick the best-matching authority id for a category, if any."""
    if not authorities:
        return None

    preferred_types = CATEGORY_TO_AUTHORITY_TYPES.get(category, [])
    for wanted in preferred_types:
        for authority in authorities:
            if authority.type.lower() == wanted:
                return authority.id

    # No type match: fall back to the first available authority.
    return authorities[0].id

File: apps/ai-service/src/providers/routing.ts
import type { Authority, ReportCategory } from "../types.js";

// Preferred authority type for each report category, best match first.
const CATEGORY_TO_AUTHORITY_TYPES: Record<ReportCategory, string[]> = {
  flood: ["fire", "municipal", "utilities"],
  accident: ["police", "health", "fire"],
  infrastructure: ["municipal", "utilities"],
  gov_delay: ["municipal"],
  safety: ["police"],
  other: ["municipal"],
};

/** Pick the best-matching authority id for a category, if any. */
export function suggestAuthorityId(
  category: ReportCategory,
  authorities: Authority[],
): string | null {
  if (authorities.length === 0) return null;

  for (const wanted of CATEGORY_TO_AUTHORITY_TYPES[category] ?? []) {
    const match = authorities.find((a) => a.type.toLowerCase() === wanted);
    if (match) return match.id;
  }

  // No type match: fall back to the first available authority.
  return authorities[0].id;
}
